modules/misc/gtk.nix

```diff
@@ -92,6 +92,18 @@ in {
             <filename>~/.gtkrc-2.0</filename>.
           '';
         };
+
+        configLocation = mkOption {
+          type = types.path;
+          apply = p:
+            if hasPrefix "/" p then p else "${config.home.homeDirectory}/${p}";
```

Comment:
I think this is unnecessary since `types.path` already implies that the path is absolute. See https://github.com/NixOS/nixpkgs/blob/dc68eb58bb8d4dc3cfb8896932e6146c45579932/lib/types.nix#L343-L347

If you want to keep the possibility of specifying a relative path, which seems reasonable, then the type will have to be `types.str`.